cmake/SanitizerOptions.cmake

```
@@ -0,0 +1,13 @@
+if(ENABLE_SANITIZER)
+  message(STATUS "Enabling address sanitizer (ASan).")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
```

On setting the flag: we could use the other cmake module that's in this folder, which will check to see if the flag is supported by the compiler being used: https://github.com/valhalla/valhalla/blob/d89b886222ee5757655f5da62a337a589b51c0db/cmake/HandleLibcxxFlags.cmake#L231. This will make it easier to support the build across platforms.

---

Do you suggest sth like 
```suggestion
  append_flags_if_supported(CMAKE_SHARED_LINKER_FLAGS -fsanitize=address)
```
? (I understand this is the most proper one here)
I generally like the approach but what disturbs me is that this function (and actually all the rest) produces no indication if the flag is not supported. I'd really rather have some warning in such case.
WDYT @mookerji? cc @PureW

---

Oh, I agree that there should definitely be a warning produced by the functions/macros in that CMake file. I just tried adding one and got a warning: 

```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0d7bb6d4a..8ade1b5c0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -63,6 +63,7 @@ endif()

 function (cxx_add_warning_flags target)
   target_add_compile_flags_if_supported(${target} PRIVATE -Wall -Wextra)
+  target_add_compile_flags_if_supported(${target} PRIVATE -Wtotallyfakeflag)
   if (ENABLE_WERROR)
    target_add_compile_flags_if_supported(${target} PRIVATE -Werror)
   endif()
```

```
$ cmake -DENABLE_PYTHON_BINDINGS=OFF -DENABLE_NODE_BINDINGS=OFF -DCMAKE_BUILD_TYPE=Debug -DENABLE_SANITIZER=OFF -DENABLE_COMPILER_WARNINGS=ON -DENABLE_WERROR=OFF -DENABLE_COVERAGE=OFF ../
-- Configuring in debug mode
-- Using ccache to speed up incremental builds
-- Using pbf headers from /usr/local/include
-- Using pbf libs from /usr/local/lib/libprotobuf.dylib
-- Using pbf release libs from
-- Using pbf debug libs from /usr/local/lib/libprotobuf.dylib
-- Using pbf-lite
-- Found SQLite3: /usr/local/opt/sqlite3/lib/libsqlite3.dylib
-- Enabling compiler warning flags
-- Performing Test LIBCXX_SUPPORTS_WALL_FLAG
-- Performing Test LIBCXX_SUPPORTS_WALL_FLAG - Success
-- Performing Test LIBCXX_SUPPORTS_WEXTRA_FLAG
-- Performing Test LIBCXX_SUPPORTS_WEXTRA_FLAG - Success
-- Performing Test LIBCXX_SUPPORTS_WTOTALLYFAKEFLAG_FLAG
-- Performing Test LIBCXX_SUPPORTS_WTOTALLYFAKEFLAG_FLAG - Failed
```

---

Wow, didn't know the check function also prints something.
Implemented this part according to the suggested approach, hope I got it correctly. PTAL